sophus/so3.hpp

```diff
@@ -257,16 +256,18 @@ class SO3Base {
     /// Representation through Encapsulation of Manifolds"
     /// Information Fusion, 2011
 
-    if (n < Constants<Scalar>::epsilon()) {
-      /// If quaternion is normalized and n=0, then w should be 1;
-      /// w=0 should never happen here!
+    if (squared_n < Constants<Scalar>::epsilon()) {
```

Comment:
I agree with @NikolausDemmel, lets keep the threshold where it is, e.g. also epsilon square, here and below.